src/Rules/Keywords/RequireFileExistsRule.php

```diff
@@ -0,0 +1,135 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Keywords;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Include_;
+use PHPStan\Analyser\Scope;
+use PHPStan\File\FileHelper;
+use PHPStan\Rules\IdentifierRuleError;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\ShouldNotHappenException;
+use function array_merge;
+use function dirname;
+use function explode;
+use function get_include_path;
+use function is_file;
+use function sprintf;
+use function str_replace;
+use const PATH_SEPARATOR;
+
+/**
+ * @implements Rule<Include_>
+ */
+final class RequireFileExistsRule implements Rule
+{
+
+	public function __construct(private string $currentWorkingDirectory)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return Include_::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$errors = [];
+		$paths = $this->resolveFilePaths($node, $scope);
+
+		foreach ($paths as $path) {
+			if ($this->doesFileExist($path, $scope)) {
+				continue;
+			}
+
+			$errors[] = $this->getErrorMessage($node, $path);
+		}
+
+		return $errors;
+	}
+
+	/**
+	 * We cannot use `stream_resolve_include_path` as it works based on the calling script.
+	 * This method simulates the behavior of `stream_resolve_include_path` but for the given scope.
+	 * The priority order is the following:
+	 * 	1. The current working directory.
+	 * 	2. The include path.
+	 *  3. The path of the script that is being executed.
+	 */
+	private function doesFileExist(string $path, Scope $scope): bool
+	{
+		$directories = array_merge(
+			[$this->currentWorkingDirectory],
+			explode(PATH_SEPARATOR, get_include_path()),
+			[dirname($scope->getFile())],
+		);
+
+		foreach ($directories as $directory) {
+			if ($this->doesFileExistForDirectory($path, $directory)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private function doesFileExistForDirectory(string $path, string $workingDirectory): bool
+	{
+		$fileHelper = new FileHelper($workingDirectory);
+		$normalisedPath = $fileHelper->normalizePath($path);
+		$absolutePath = $fileHelper->absolutizePath($normalisedPath);
+
+		return is_file($absolutePath);
+	}
+
+	private function getErrorMessage(Include_ $node, string $filePath): IdentifierRuleError
+	{
+		$message = 'Path in %s() "%s" is not a file or it does not exist.';
+
+		switch ($node->type) {
+			case Include_::TYPE_REQUIRE:
+				$type = 'require';
```

Comment:
One last simple change: I worry the identifier cannot be statically inferred here because of `str_replace`. We need that for https://github.com/phpstan/phpstan/actions/workflows/extract-identifiers.yml which updates https://phpstan.org/error-identifiers.

Please assign `$identifier = ...` in the switch in each case, and use that in RuleErrorBuilder.

Then I'll merge this. Thank you for the effort!

Reply:
Fixed in https://github.com/phpstan/phpstan-src/pull/3294/commits/e861017a0793a3d8f6dc5b7192c6d4d7c59d217f.

Shouldn't this also run as a quality check before merging a PR?